src/main.cc

```diff
@@ -178,6 +178,25 @@ signal_cb(ev_loop *loop, struct ev_signal *w, int revents)
 	tarantool_exit(0);
 }
 
+static void
+signal_int_cb(ev_loop *loop, struct ev_signal *w, int revents)
+{
+    (void) loop;
+    (void) w;
+    (void) revents;
+
+    if (pid_file)
+        say_crit("got signal %d - %s", w->signum, strsignal(w->signum));
+
+    /* Regenerate the prompt on a newline. */
+    printf("\n");
+    rl_on_new_line();
+
+    /* Clear the previous text. */
+    rl_replace_line("", 0);
+    rl_redisplay();
```

Comment:
Observed that we don't exit from the search mode (started by Ctrl+R) at Ctrl+C.

Reply:
It seems, that it should be something like `rl_restore_prompt()` + `RL_UNSETSTATE(RL_STATE_ISEARCH|RL_STATE_NSEARCH|RL_STATE_SEARCH)`, but it does not work well for several Ctrl+C hits for me. Need to dig a bit deeper.

Reply:
Is it more preferably to cancel search mode with Ctrl+C or let it discard the current input in search line and return the new one for input? For now it has the second behavior. Bash has the first one, I guess it's expected that Tarantool console behaves like this too.

Reply:
I would leave the search mode (and discard the search line) and go to the normal input. As good we'll mimic the usual interactive shell behaviour as good it will be felt by users.

Reply:
Done. It leaves the search mode with `^C` output. There is the same behavior in bash.